### PR TITLE
Fix compatibility with PIL

### PR DESCRIPTION
**Pull request by tonysyu**
It looks like there was added with the intention of supporting PIL compatibility (vs pillow), but it doesn't look like this was fully tested. (At least it breaks on my system.)

### CODE REVIEW

**Comment by jni** [comment]
@tonysyu can you elaborate a bit on what breaks, and whether we could have tested something differently to catch it? Is it PIL or Pillow that's broken?

**Comment by tonysyu** [comment]
I get an `AttributeError` when running with PIL.

I think Pillow added `tobytes`/`frombytes` when they added support for Python 3.0. Notice the calls to `tostring` (near the code changed in this PR) for backwards compatibility with PIL; these should have been accompanied by calls to `fromstring` rather than `frombytes`. I'm guessing that all of the TravisCI instances are running Pillow rather than PIL, otherwise this would break.

**Comment by tonysyu** [comment]
Hmm... I guess the `tostring` is coincidental, since those are numpy arrays. But nevertheless, AFAICT PIL does not have a `frombytes` method.

**Comment by jni** [comment]
Well, I just noticed that we actually specify Pillow as a dependency now, so I don't know whether it's worth running a separate instance using PIL in Travis. (@stefanv?) But assuming the build passes, it's good to have PIL compatibility for crazy people who ignore dependency lists! ;)

**Comment by tonysyu** [comment]
Hmm, ... it looks like `tobytes` for numpy arrays is *very* new. That conversion should be tested independently of the `frombytes`/`fromstring` methods in PIL/pillow. (See most recent commit)

I think it'd be a bit of work to test both Pillow and PIL in Travis CI, since it doesn't look like the travis setup tries to do anything special with versions of dependencies (other than Python versions).

**Comment by tonysyu** [comment]
Hmm, ... it looks like `tobytes` for numpy arrays is *very* new. That conversion should be tested independently of the `frombytes`/`fromstring` methods in PIL/pillow. (See most recent commit)

I think it'd be a bit of work to test both Pillow and PIL in Travis CI, since it doesn't look like the travis setup tries to do anything special with versions of dependencies (other than Python versions).

File: skimage/io/_plugins/pil_plugin.py [hunk 1]
```diff
@@ -176,20 +176,23 @@ def ndarray_to_pil(arr, format_str=None):
...
         except AttributeError:
-            im.frombytes(arr.tostring(), 'raw', mode)
-
+            im.fromstring(array_buffer, 'raw', mode)  # PIL 1.1.7
     else:
+        image_shape = (arr.shape[1], arr.shape[0])
         try:
-            im = Image.frombytes(mode, (arr.shape[1], arr.shape[0]),
-                                 arr.tobytes())
+            im = Image.frombytes(mode, image_shape, array_buffer)
```
Comment (jni) [inline skimage/io/_plugins/pil_plugin.py]: Wow, are the function signatures for `Image.frombytes` and `Image.new().frombytes` really so different??? Incredible!

**Comment by jni** [comment]
This looks much more robust than before! =)

File: skimage/io/_plugins/pil_plugin.py (hunk 1 above)
Comment (tonysyu) [inline skimage/io/_plugins/pil_plugin.py]: Yeah,... that's a pretty bizarre API.

**Comment by tonysyu** [comment]
Coveralls: You're really making me look bad!

**Comment by jni** [comment]
Holy moly!!! LOL Well Travis looks like it *has* run all the tests, so I'm just gonna merge this. Thanks @tonysyu!

**Pull request merged**